### PR TITLE
Feature/jackalope 2 0 upgrade

### PR DESCRIPTION
**Pull request by mamazu**
Fixing up the test (see comments for details)

### CODE REVIEW

File: src/Sulu/Component/Content/Repository/ContentRepository.php [hunk 1]
```diff
@@ -527,7 +527,7 @@ private function getQueryBuilder($locale = null, $locales = [], ?UserInterface $
         return $queryBuilder;
     }
 
-    private function getPropertyName($propertyName, $locale)
+    private function getPropertyName(string $propertyName, $locale): string
```
Comment (mamazu) [inline src/Sulu/Component/Content/Repository/ContentRepository.php:530]: Not a BC break as it would crash when the method does not return a string (which is what the failing tests complain about. 
Parameter is always a string (see usage above).

File: src/Sulu/Component/Content/Tests/Unit/Repository/ContentRepositoryTest.php [hunk 2]
```diff
@@ -110,6 +111,10 @@ public function setUp(): void
         $this->sessionManager = $this->prophesize(SessionManagerInterface::class);
         $this->documentManager = $this->prophesize(DocumentManagerInterface::class);
         $this->propertyEncoder = $this->prophesize(PropertyEncoder::class);
+        $this->propertyEncoder->localizedContentName(Argument::type('string'), Argument::type('string'))
+            ->will(fn ($args) => 'i18n-'.$args[1].'-'.$args[0]);
+        $this->propertyEncoder->systemName('order')->willReturn('order');
```
Comment (mamazu) [inline src/Sulu/Component/Content/Tests/Unit/Repository/ContentRepositoryTest.php:116]: This was missing from the test and just was never caught in the older versions of the Query as PHPCR didn't check for null arguments

File: src/Sulu/Component/DocumentManager/Tests/Unit/NodeHelperTest.php [hunk 3]
```diff
@@ -79,35 +80,37 @@ public function testMoveWithDestinationName(): void
...
+        /** @var ObjectProphecy<Workspace> $workspace */
         $workspace = $this->prophesize(Workspace::class);
-        $this->session->getWorkspace()->willReturn($workspace->reveal());
+    $this->session->getWorkspace()->willReturn($workspace->reveal());
+        $identifier = UUIDHelper::generateUUID();
 
         $destinationNode = $this->prophesize(NodeInterface::class);
         $destinationNode->getPath()->willReturn('/path/to/some/other/node');
-        $this->session->getNodeByIdentifier('uuid')->willReturn($destinationNode->reveal());
+        $this->session->getNodeByIdentifier($identifier)->willReturn($destinationNode->reveal());
```
Comment (mamazu) [inline src/Sulu/Component/DocumentManager/Tests/Unit/NodeHelperTest.php:90]: As discussed in the main PR: This only worked because the prophecy extension never checked the method calls. fixed them now.

File: src/Sulu/Component/Util/SuluNodeHelper.php [hunk 4]
```diff
@@ -314,25 +314,27 @@ public function getTranslatedProperty($property, $locale, $prefix = null)
...
+                if ($previous){
+                    return $previousNode;
+                }
+                $children->next();
+                return $children->current();
             }
 
             $previousNode = $child;
-            \next($children);
+            $children->next();
```
Comment (mamazu) [inline src/Sulu/Component/Util/SuluNodeHelper.php:337]: The tests are now using an iterator which makes the syntax slightly different. But it should still work the same.

File: src/Sulu/Component/Util/Tests/Unit/SuluNodeHelperTest.php [hunk 5]
```diff
@@ -254,19 +254,20 @@ public function testSiblingNodes(): void
...
             ->method('getNodes')
-            ->will($this->returnValue([
-                $node1, $node2, $node3,
-            ]));
+            ->will($this->returnValue($iterator));
 
         $res = $this->helper->getNextNode($node2);
         $this->assertSame($node3->getPath(), $res->getPath());
 
+        $iterator->rewind();
```
Comment (mamazu) [inline src/Sulu/Component/Util/Tests/Unit/SuluNodeHelperTest.php:270]: We need to rewind the iterators in between usages as Phpunit will not create the iterator again when the `getNodes` method  is called a second time.

**Review by alexander-schranz — APPROVED** [review]
